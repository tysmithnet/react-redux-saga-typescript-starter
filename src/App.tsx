import * as React from "react";
import { applyMiddleware, createStore } from "redux";
import createSagaMiddleware from "redux-saga";
import rootSaga from "./sagas";
import something from "./something";

export function reducer(state: any, action: any) {
  console.log("reducer called");
}

const sagaMiddleware = createSagaMiddleware();
const store = createStore(reducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(rootSaga);

export default class App extends React.Component<{}, {}> {
  public render() {
    return <h1>hi!</h1>;
  }
}
