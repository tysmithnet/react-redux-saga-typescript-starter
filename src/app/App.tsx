import * as React from "react";
import { hot } from "react-hot-loader";
import { connect } from "react-redux";
import Auth from "../auth/Auth";
import { Menu } from "../menu/Menu";
import { IRootState } from "../root.state";
import { IProps } from "./domain";

export class App extends React.Component<IProps> {
  public render() {
    let component: React.ReactNode = null;
    if (this.props.user) {
      component = <h1>Welcome!! {this.props.user.name}1!!</h1>;
    } else {
      component = <Auth dispatch={this.props.dispatch} />;
    }

    return (
      <div>
        {component}
        <Menu name="men11uuu" />
      </div>
    );
  }
}

function mapStateToProps(state: IRootState): any {
  return {
    user: state.auth.user,
  };
}

const connectedComponent = connect(mapStateToProps)(hot(module)(App));
export default connectedComponent;
