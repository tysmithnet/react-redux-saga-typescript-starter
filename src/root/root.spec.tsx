  
import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import {App} from "./App";

configure({ adapter: new (Adapter as any)() });

test("Sanity", () => {
    const checkbox = shallow(<App labelOn={"on"} labelOff={"off"} />);
    expect(checkbox.text()).toEqual("off");
    checkbox.find("input").simulate("change");
    expect(checkbox.text()).toEqual("on");
    expect(shallow).toMatchSnapshot();
});