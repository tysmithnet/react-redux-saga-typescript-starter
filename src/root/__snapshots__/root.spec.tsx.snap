// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sanity 1`] = `[Function]`;
